### PR TITLE
Update SalesFormsPref fields on Preferences model

### PR DESCRIPTION
**Pull request by drewish**
I needed to check the AutoApplyCredit value and realized it wasn't present in the model. I went through [the docs](https://developer.intuit.com/app/developer/qbo/docs/api/accounting/most-commonly-used/preferences#the-preferences-object) for that object and pulled all the current  fields in the SalesFormsPref section.

### CODE REVIEW

File: lib/quickbooks/model/preferences.rb [hunk 1]
```diff
@@ -28,9 +28,28 @@ def self.create_preference_class(*attrs, &block)
...
+        AllowShipping?
+        AutoApplyCredit?
+        CustomField?
+        CustomTxnNumbers?
+        DefaultCustomerMessage
+        DefaultDiscountAccount?
+        DefaultShippingAccount?
+        DefaultTerms
+        EmailCopyToCompany?
+        EstimateMessage
```
Comment (drewish) [inline lib/quickbooks/model/preferences.rb]: Before we had this flagged as a Boolean but the docs show it as a string.

File: lib/quickbooks/model/preferences.rb [hunk 2]
```diff
@@ -28,9 +28,28 @@ def self.create_preference_class(*attrs, &block)
...
+        ETransactionAttachPDF?
+        ETransactionEnabledStatus
```
Comment (drewish) [inline lib/quickbooks/model/preferences.rb]: Before we had this as a Boolean but the docs show it as an `ETYransactionEnabledStatusEnum`

File: lib/quickbooks/model/preferences.rb [hunk 3]
```diff
@@ -28,9 +28,28 @@ def self.create_preference_class(*attrs, &block)
...
+        ETransactionPaymentEnabled?
+        IPNSupportEnabled?
+        SalesEmailBcc
+        SalesEmailCc
+        UsingPriceLevels?
+        UsingProgressInvoicing?
```
Comment (drewish) [inline lib/quickbooks/model/preferences.rb]: I put these into a sorted order to make it easier to diff later changes.

File: lib/quickbooks/model/preferences.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@ class Preferences < BaseModel
       XML_COLLECTION_NODE = "Preferences"
       XML_NODE            = "Preferences"
       REST_RESOURCE       = 'preferences'
-      MINORVERSION = 21
+      MINORVERSION = 32
```
Comment (drewish) [inline lib/quickbooks/model/preferences.rb]: Bumping this gets us access to the `UsingProgressInvoicing` field.

**Comment by ruckus** [comment]
Thank you!

**Comment by drewish** [comment]
@ruckus You could close https://github.com/ruckus/quickbooks-ruby/pull/483 now too if you wanted.

**Pull request merged**